lib/nebulex/adapters/mnesia/cluster.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Nebulex.Adapters.Mnesia.Cluster do
+  @moduledoc """
+  This module manages Mnesia cluster nodes and table copies.
+  It monitors node up and down events to ensure that table copies are
+  added or removed as needed.
+  """
+
+  use GenServer
+
+  @table_attrs ~w[key value touched ttl]a
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  def init(opts) do
+    :mnesia.start()
+    :net_kernel.monitor_nodes(true)
+    cache = Keyword.fetch!(opts, :cache)
+    table = Keyword.get(opts, :table, cache)
+    master = Keyword.get(opts, :master_node, false)
+    if master, do: create_schema_and_table(table)
+    {:ok, %{table: table, master: master}}
+  end
+
+  def handle_info({:nodeup, _node}, %{table: table, master: master} = state) do
+    unless master, do: add_table_copy(table)
+    {:noreply, state}
+  end
+
+  def handle_info({:nodedown, _node}, state) do
+    {:noreply, state}
+  end
+
+  defp create_schema_and_table(table) do
+    :mnesia.stop()
+    :mnesia.create_schema([node()])
```

Comment:
To be more elixir idiomatic, consider the use of  Node.self() instead of node, they are the same, but Elixir preferred is the first one.

Reply:
Fixed, please validate.